clocksilencer.ino

```diff
@@ -76,28 +77,24 @@ void loop() {
   }
   struct tm timeinfo = rtc.getTimeStruct();
   esp_sleep_enable_timer_wakeup(500000);  //in microseconds, but not the full second (1000000)
-  if (bootCount > 2147483645) {
-    bootCount = 2;  // when bootcount approaches to integer max, reset it
-  }
   Serial.println("Deepsleep");
   esp_deep_sleep_start();
+  //delay(1000);
 }
 void ringer(int myhours2) {
-  Serial.print("Ringing:");
-  Serial.println(myhours2);
   if (myhours2 > 12) {
-    myhours2 - 12;
+    myhours2=myhours2 - 12;
```

Comment:
Or maybe
`if (myhours2 > 12) {
    myhours2 -= 12;} `

Reply:
The reason is because:

```
myhours2 - 12;
```

does not change the value of the variable. it is an expression (so e.g. for `13` the result will be `1`), but the result is not stored anywhere.

By having 

```
myhpurs2 = myhours2 - 12;
```

we save the new value into the `myhours2` variable.

Reply:
So an assigment operator - = is needed Here. Got it.